pkg/config/setup/config.go

```diff
@@ -2357,6 +2361,17 @@ func sanitizeExternalMetricsProviderChunkSize(config pkgconfigmodel.Config) {
 	}
 }
 
+func toggleDefaultPayloads(config pkgconfigmodel.Config) {
+	// Disables metric data submission (including Custom Metrics) so that hosts stop showing up in Datadog.
+	// Used namely for Error Tracking Standalone where it is not needed.
+	if !config.GetBool("core_agent.enabled") {
+		config.BindEnvAndSetDefault("enable_payloads.events", false)
+		config.BindEnvAndSetDefault("enable_payloads.series", false)
+		config.BindEnvAndSetDefault("enable_payloads.service_checks", false)
+		config.BindEnvAndSetDefault("enable_payloads.sketches", false)
```

Comment:
This redeclares the config key with a different default rather than override their value, I believe this is the correct way to do it
```suggestion
		config.Set("enable_payloads.events", false, pkgconfigmodel.SourceAgentRuntime)
		config.Set("enable_payloads.series", false, pkgconfigmodel.SourceAgentRuntime)
		config.Set("enable_payloads.service_checks", false, pkgconfigmodel.SourceAgentRuntime)
		config.Set("enable_payloads.sketches", false, pkgconfigmodel.SourceAgentRuntime)
```